enhancements/orange_okt_proposition.md

```diff
@@ -0,0 +1,326 @@
+
+I like the idea to have a clear view on the steps I defined previously, so I'll complete my work with the OperatorSDK and the OKT addon.
+
+OKT comes with a statemachine feature that should help in defining these steps and let me focus on the code I need to implement at each step. 
+To allow this, OKT provides:
+
+  -  a sidecar for my application to help me to get my database status and launch actions on it asynchronously.
+  -  an utility to modelize my graph of appication states into my CRD
+  -  a GO type to implement this graph and transition rules that condition how I validate the transition from one step to another
+
+In my CR I set the wished state (i.e. Servicing) I want to reach, while the current application state (i.e. is maintained in the CR status with a new Condition).
+
+Once the application added to the OKT registry (like any other resource), the OKT Reconciler knows that it has to manage this  resource as follow:
+
+  - on Start: Create() it!
+  - on End: Delete() it!
+  - on any other state: Update it!
+
+As any other resource, it put in place an idempotent mecanism and detect changes (and thus will do nothing during a Reconciliation if there's nothing new). Here what will trigger a change:
+
+  - a state change (in App LC Graph) due to a CR modification
+  - a state change from the observation of a change at the application level. This observability should be implemented by an application sidecar container or a usable function in the application container itself. 
+ 
+ A state change (in the App LC  Graph) is handled asynchronously to not impact the Controller with a too long task. On such case (long task) 1 or more requeueing orders are left to wait for the observable change once done. 
+
+It also maintain a Status condition in the CR that reflect the application current state and errors if any.
+
+To sum up:
+
+  - an application lifecycle is managed like an infrastucture resource from OKT's point of view, 
+- a clear view on what is implemented in term of application lifecycle is provided thanks to the App LC Graph described by the CRD
+  - Having all the operators in an organization built upon the same model should help human (or intelligent automates) operators to deal with several kind of K8S operators.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Today, the Story 1 is fully operational and implementable with the current version of OKT. A partial implementation detail can be provided by the MemcachedOperatorSampleWithOKT repo code.
+
+The architecture to render story 2 implementable is not yet fully completed and we are wondering if this approach make sense or not.
+
+The OKT library is a GO module that depends on the OperatorSDK, more specifically on the sigs and k8s.io modules aligned on those used by the OperatorSDK.
+
+Upgrading the OperatorSDK version means upgrading OKT, it would be less impacting if OKT was integrated in the OperatorSDK as an internal tool box. 

```

Comment:
I'm not sure it has to be in lock step version for version. I think OKT could have it's own versioning scheme and still be compatible with Operator SDK. I think if we made it an internal toolkit, that would mean we'd have to expose a public library from within the Operator SDK repo which we didn't want to do. I think OKT could have a life of its own alongside Operator SDK. I could potentially see it merged with [operator-lib](https://github.com/operator-framework/operator-lib/) or vice versa.